helm/grafana/templates/grafana-deployment.yaml

```diff
@@ -43,10 +44,14 @@ spec:
         volumeMounts:
         - name: grafana-storage
           mountPath: /var/lib/grafana
-      {{- if .Values.mountGrafanaConfig }}
+        {{- if .Values.mountGrafanaConfig }}
         - name: grafana-config
           mountPath: /etc/grafana
-      {{- end }}
+        {{- end }}
+        {{- if $useCustomConfig }}
+        - name: grafana-config
+          mountPath: /etc/grafana/conf        
```

Comment:
Put this one directory higher so that it does not clash with the existing configuration, and changes to the grafana.ini can be done using the extraEnv by defining `GF_...`